packages/react/toast/src/Toast.tsx

```diff
@@ -794,21 +778,47 @@ const ToastClose = React.forwardRef<ToastCloseElement, ToastCloseProps>(
   (props: ScopedProps<ToastCloseProps>, forwardedRef) => {
     const { __scopeToast, ...closeProps } = props;
     const interactiveContext = useToastInteractiveContext(CLOSE_NAME, __scopeToast);
-    return interactiveContext.isInteractive ? (
-      <Primitive.button
-        type="button"
-        {...closeProps}
-        ref={forwardedRef}
-        onClick={composeEventHandlers(props.onClick, interactiveContext.onClose)}
-      />
-    ) : null;
+
+    return (
+      <ToastAnnounceExclude asChild>
+        <Primitive.button
+          type="button"
+          {...closeProps}
+          ref={forwardedRef}
+          onClick={composeEventHandlers(props.onClick, interactiveContext.onClose)}
+        />
+      </ToastAnnounceExclude>
+    );
   }
 );
 
 ToastClose.displayName = CLOSE_NAME;
 
 /* ---------------------------------------------------------------------------------------------- */
 
+type ToastAnnounceExcludeElement = React.ElementRef<typeof Primitive.div>;
+interface ToastAnnounceExcludeProps extends PrimitiveDivProps {
+  altText?: string;
+}
+
+const ToastAnnounceExclude = React.forwardRef<
```

Comment:
I added a new comp with idea of colocating `getAnnounceTextContent` near it in the file, I haven't moved the code yet to spare the diff but let me know if you think this is overkill in general.

Reply:
I'm not fussed about `getAnnounceTextContent` being near it in the file really.
But seeing the data attributes here in one place, rather than spread out in both places's implementation is a good thing to keep, so I guess we get the bonus of the collocation if you want to move it before merging the PR.